### PR TITLE
PADV-961 feat: Add instructor assignment section

### PR DESCRIPTION
**Pull request by AuraAlba**
### Ticket
https://agile-jira.pearson.com/browse/PADV-961

### Description
Add instructor assignment section

### Changes made
Add instructor assignment section
Move getClassesByInstitution to common folder
Replace moment for date-fns required for the calendar component
Update tests
Define renderWithProviders for redux test to avoid import of provider in every test

### Screenshots
Screenshot from 2024-01-25 18-41-32

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1980/dashboard

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
-// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (anfbermudezme) [inline .eslintrc.js:1]: Why this is working now? @AuraAlba

File: src/features/Dashboard/InstructorAssignSection/index.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  let idInstitution = '';
+  // eslint-disable-next-line no-unused-expressions
+  stateInstitution.length > 0 ? idInstitution = stateInstitution[0].id : idInstitution = '';
+
+  useEffect(() => {
+    dispatch(fetchClassesData(idInstitution));
+  }, [idInstitution]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    // Show in home only two first classes
```
Comment (anfbermudezme) [inline src/features/Dashboard/InstructorAssignSection/index.jsx:26]:
```suggestion
    // Display only the first 'NumberOfClasses' on the homepage.
```

File: src/features/Dashboard/InstructorAssignSection/index.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if (classesData.length > 2) {
```
Comment (anfbermudezme) [inline src/features/Dashboard/InstructorAssignSection/index.jsx:27]: What do you think set the number 2 as a variable? @AuraAlba 
```suggestion
    if (classesData.length > NumberOfClasses) {
```

File: src/features/Dashboard/InstructorAssignSection/index.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      setClassCards(classesData.slice(0, 2));
```
Comment (anfbermudezme) [inline src/features/Dashboard/InstructorAssignSection/index.jsx:28]:
```suggestion
      setClassCards(classesData.slice(0, NumberOfClasses));
```

File: src/features/Dashboard/InstructorAssignSection/index.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      setClassCards(classesData);
+    }
+  }, [classesData]);
+
+  return (
+    <Row>
+      <Col xs="12">
+        <h4 className="title-instr-assign">Instructor assignment</h4>
+        {classCards.map(classInfo => <ClassCard data={classInfo} />)}
+        {classesData.length > 2 && (
```
Comment (anfbermudezme) [inline src/features/Dashboard/InstructorAssignSection/index.jsx:39]:
```suggestion
        {classesData.length > NumberOfClasses && (
```

File: src/test-utils.jsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  await thunk(dispatch, getState);
+};
+
+export function renderWithProviders(
+  ui,
+  {
+    preloadedState = {},
+    // Automatically create a store instance if no store was passed in
+    store = initializeStore(preloadedState),
+    ...renderOptions
```
Comment (anfbermudezme) [inline src/test-utils.jsx:17]:
```suggestion
    ...renderOptions,
```
Trailing comma.

File: .eslintrc.js (hunk 1 above)
Comment (AuraAlba) [reply .eslintrc.js:1]: I added this file in the list of eslintignore so this line is not necessary

File: src/features/Dashboard/InstructorAssignSection/index.jsx (hunk 2 above)
Comment (AuraAlba) [reply src/features/Dashboard/InstructorAssignSection/index.jsx:26]: Why number of classes? in the home page will display the two first classes without instructor assign

Comment (AuraAlba) [reply src/features/Dashboard/InstructorAssignSection/index.jsx:26]: got it, is a variable

File: src/test-utils.jsx (hunk 6 above)
Comment (AuraAlba) [reply src/test-utils.jsx:17]: This change makes failed redux test: Unexpected trailing comma after rest element. So I won't add it

File: src/features/Dashboard/InstructorAssignSection/index.jsx (hunk 3 above)
Comment (AuraAlba) [reply src/features/Dashboard/InstructorAssignSection/index.jsx]: Done

File: src/features/Dashboard/InstructorAssignSection/index.jsx (hunk 4 above)
Comment (AuraAlba) [reply src/features/Dashboard/InstructorAssignSection/index.jsx]: Done

File: src/features/Dashboard/InstructorAssignSection/index.jsx (hunk 5 above)
Comment (AuraAlba) [reply src/features/Dashboard/InstructorAssignSection/index.jsx]: Done

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**